src/mdns/mdns_mdnssd.cpp

```diff
@@ -998,7 +995,7 @@ void PublisherMDnsSd::ServiceSubscription::GetAddrInfo(uint32_t aInterfaceIndex)
 
     otbrLogInfo("DNSServiceGetAddrInfo %s inf %d", mInstanceInfo.mHostName.c_str(), aInterfaceIndex);
 
-    DNSServiceGetAddrInfo(&mServiceRef, /* flags */ 0, aInterfaceIndex,
+    DNSServiceGetAddrInfo(&mServiceRef, /* flags */ kDNSServiceFlagsTimeout, aInterfaceIndex,
```

Comment:
May also mention this change in PR description.

Reply:
`kDNSServiceFlagsTimeout` allows address resolving to quit when timeout.

Reply:
nit: should we remove the `/* flags */` comment? Instance above removed this comment.
```suggestion
    DNSServiceGetAddrInfo(&mServiceRef, kDNSServiceFlagsTimeout, aInterfaceIndex,
```

Reply:
Took me a while to find this comment since it's resolved. :smile: 

Fixed.